programs/bpf_loader/src/lib.rs

```diff
@@ -219,15 +201,92 @@ pub fn create_executor_from_account(
                 ic_logger_msg!(log_collector, "Program has been closed");
                 return Err(InstructionError::InvalidAccountData);
             }
-            UpgradeableLoaderState::size_of_programdata_metadata()
+            Ok(UpgradeableLoaderState::size_of_programdata_metadata())
         } else {
             ic_logger_msg!(log_collector, "Invalid Program account");
-            return Err(InstructionError::InvalidAccountData);
+            Err(InstructionError::InvalidAccountData)
         }
+    } else {
+        Ok(0)
+    }
+}
+
+pub fn load_program_from_account(
+    feature_set: &FeatureSet,
+    compute_budget: &ComputeBudget,
+    log_collector: Option<Rc<RefCell<LogCollector>>>,
+    program: &BorrowedAccount,
+    programdata: &BorrowedAccount,
+) -> Result<(LoadedProgram, Option<CreateMetrics>), InstructionError> {
+    if !check_loader_id(program.get_owner()) {
+        ic_logger_msg!(
+            log_collector,
+            "Executable account not owned by the BPF loader"
+        );
+        return Err(InstructionError::IncorrectProgramId);
+    }
+
+    let programdata_offset = get_programdata_offset(&log_collector, program, programdata)?;
+    let programdata_size = if programdata_offset != 0 {
+        programdata.get_data().len()
     } else {
         0
     };
 
+    let mut create_executor_metrics = CreateMetrics {
+        program_id: program.get_key().to_string(),
+        ..CreateMetrics::default()
+    };
+
+    let mut register_syscalls_time = Measure::start("register_syscalls_time");
+    let loader = syscalls::create_loader(feature_set, compute_budget, false, false, false)
+        .map_err(|e| {
+            ic_logger_msg!(log_collector, "Failed to register syscalls: {}", e);
+            InstructionError::ProgramEnvironmentSetupFailure
+        })?;
+    register_syscalls_time.stop();
+    create_executor_metrics.register_syscalls_us = register_syscalls_time.as_us();
+
+    let mut load_elf_time = Measure::start("load_elf_time");
+    let loaded_program = LoadedProgram::new(
+        program.get_owner(),
+        loader,
+        0, // Fill in the deployment slot of the program
```

Comment:
TODO left.

This could be returned in `get_programdata_offset()`:
https://github.com/solana-labs/solana/blob/a349b74496f0d784d8728867e5509da0324a6f32/programs/bpf_loader/src/lib.rs#L197
And maybe rename it to `get_programdata_offset_and_deployment_slot()`.